x-pack/plugin/ilm/qa/multi-node/src/javaRestTest/java/org/elasticsearch/xpack/ilm/ExplainLifecycleIT.java

```diff
@@ -305,16 +307,23 @@ public void testStepInfoPreservedOnAutoRetry() throws Exception {
         indexRequest.setJsonEntity("{\"test\":\"value\"}");
         assertOK(client().performRequest(indexRequest));
 
-        assertBusy(() -> {
-            Map<String, Object> explainIndex = explainIndex(client(), indexName);
-            assertThat(explainIndex.get("failed_step_retry_count"), notNullValue());
-            assertThat(explainIndex.get("previous_step_info"), notNullValue());
-            assertThat((int) explainIndex.get("failed_step_retry_count"), greaterThan(0));
-            assertThat(
-                explainIndex.get("previous_step_info").toString(),
-                containsString("rollover_alias [" + aliasName + "] does not point to index [" + indexName + "]")
-            );
-        });
+        AtomicReference<Map<String, Object>> explainIndexRef = new AtomicReference<>();
+        try {
+            assertBusy(() -> {
+                Map<String, Object> explainIndex = explainIndex(client(), indexName);
+                explainIndexRef.set(explainIndex);
+                assertThat(explainIndex.get("failed_step_retry_count"), notNullValue());
+                assertThat(explainIndex.get("previous_step_info"), notNullValue());
+                assertThat((int) explainIndex.get("failed_step_retry_count"), greaterThan(0));
+                assertThat(
+                    explainIndex.get("previous_step_info").toString(),
+                    containsString("rollover_alias [" + aliasName + "] does not point to index [" + indexName + "]")
+                );
+            }, 30, TimeUnit.SECONDS);
+        } catch (AssertionError e) {
+            logger.error("Assertion failed with response: {}", explainIndexRef.get());
+            throw e;
+        }
```

Comment:
If you didn't want to catch the AssertionError (we generally try never to catch any Error classes, only Exceptions), it may be better to put a logging statement at `info` level inside of the `assertBusy()`, which would log the output each time (and show up in the logs if it fails again). Then you wouldn't need the `AtomicReference` either.

Reply:
That is how I initially implemented it however it really did pollute the logs and looked a bit confusing as it logged out during each attempt at the assertBusy.

I also thought about adding it as a message param to each assertion however those aren't lazy evaluated like a log message param so it was adding to the execution time to do the map.toString for each assertion when it was 99% of the time not actually used.

In the end I decided to optimized for execution time (I'm always conscious time == money in the cloud CI environment) and conciseness but happy to go for logging it out each time if you feel strongly about not catching errors :-)

Reply:
I think it'd be nice to avoid the error catching if possible, so I would advocate for adding the message param to the assertions you think are most likely to fail (`assertBusy`'s back-off gets longer and longer, so I don't think it'd be too bad). Then it's only returned if the entire thing fails.

Reply:
Switched to using a message param on the assertions and saving the message to a var so it's only computed once per run